### PR TITLE
test: for non-ASCII characters (#9728)

### PR DESCRIPTION
**Pull request by moyodiallo**
See #9728.

### CODE REVIEW

**Comment by moyodiallo** [comment]
Why the check failing only on macos with this error https://github.com/ocaml/dune/actions/runs/7728961114/job/21071024492?pr=9877. AFAIK even though there's no execution involve with the dune file only formatted.

File: test/blackbox-tests/test-cases/formatting/non-ascii-characters.t [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  > (run foo %{bin:é})
+  File "", line 1, characters 15-16:
+  Error: The character '\195' is not allowed inside %{...} forms
+  run: not found
+  [127]
+
+  $ dune format-dune-file <<EOF
+  > (= "É" "\195\137")
+  > EOF
+  (= "\195\137" "\195\137")
```
Comment (emillon) [inline test/blackbox-tests/test-cases/formatting/non-ascii-characters.t:20]: This part is interesting to add, but not just as a formatting check. We want to make sure that this evaluates to `true`. Maybe as `test-cases/enabled_if/non-ascii.t`?

Comment (moyodiallo) [reply test/blackbox-tests/test-cases/formatting/non-ascii-characters.t:19]: Seeing the result of after parsed, the evaluation will be a comparison between string. Is it really necessary to add that evaluation ?

Comment (emillon) [reply test/blackbox-tests/test-cases/formatting/non-ascii-characters.t:19]: yes, this shows that we do the comparison after interpreting escape directives.

Comment (moyodiallo) [reply test/blackbox-tests/test-cases/formatting/non-ascii-characters.t:19]: I get it.

Comment (moyodiallo) [reply test/blackbox-tests/test-cases/formatting/non-ascii-characters.t:19]: Done in https://github.com/ocaml/dune/pull/9877/commits/9d2d404e7b8ba44b4b302edac91b8102cff1506a.

File: test/blackbox-tests/test-cases/enabled_if/non-ascii.t [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Afer the parsing non-ascii characters could be compared through expressions.
+Linked to the issue #9728
+
+  $ echo '(lang dune 3.0)' > dune-project
+  $ cat >dune <<"EOF"
+  > (rule
+  >  (alias default)
```
Comment (emillon) [inline test/blackbox-tests/test-cases/enabled_if/non-ascii.t:7]: nice trick to use the default alias here

**Comment by emillon** [comment]
Thanks. I removed the superfluous test

File: test/blackbox-tests/test-cases/enabled_if/non-ascii.t (hunk 2 above)
Comment (moyodiallo) [reply test/blackbox-tests/test-cases/enabled_if/non-ascii.t:7]: Thanks, but It was inspired by `/test/blackbox-tests/test-cases/enabled_if/read-macro.t`.

**Pull request merged**